### DISCUSSION TITLE
Higher precision for temperature readings

### DISCUSSION THREAD

**Discussion by bgunton** [PeteRager/lennoxs30#233] [Q&A] [answered]
Does the data from the device provide temperature with more precision than one degree Fahrenheit, and if so, is it possible to have that come through on the component?

Also, fantastic work on this component, many thanks.

**Comment by PeteRager** [comment, marked as answer]
The temperatures from the zone reporting from the Lennox device are increments of 1 degree F or 0.5 degree C.  Basically whatever value Lennox sends is what gets passed to HASS. 

That said, some of the diagnostic data has a higher resolution, for example the diagnostic sensor for outdoor temperature ou_ambient_temp is provided in increments of 0.1 degree F vs 1 degree increment of the outdoor_temperature sensor.  So I you could enable diagnostics and see if there is any data with better resolution.  I haven't seen any, but then again each model of equipment has its unique set.

**Comment by bgunton** [comment]
Great, I'll dig into the diagnostics data and see if I can find some more precise readings. Is there any harm to having diagnostics on indefinitely? Thank you!

**Comment by PeteRager** [comment]
I've been running with them on for a year continuously.  The only issues reported are related to using diagnostics when the s30 is internet connected, but I've also had people using them with the internet connected just fine.  Lennox has updated their software so those old issues may not still exist.

Either way I'd look at this documentation to understand what could go wrong.

https://github.com/PeteRager/lennoxs30/blob/master/docs/diagnostics.md
